Student-Unity/src/main/java/csed/swe/studentunity/Logic/RegistrationService.java

```diff
@@ -1,37 +1,61 @@
 package csed.swe.studentunity.Logic;
 
 import csed.swe.studentunity.model.RegistrationRequest;
+import csed.swe.studentunity.model.UnverifiedUser;
 import csed.swe.studentunity.model.User;
+import csed.swe.studentunity.model.VerificationRequest;
 import lombok.AllArgsConstructor;
 import org.springframework.stereotype.Service;
 
+import java.util.Optional;
+
 @AllArgsConstructor
 @Service
 //@Transactional
 public class RegistrationService {
 
     private final UserService userService;
+    private final UnverifiedUserService unverifiedUserService;
+    private final EmailService emailService;
 
     public String addUser(RegistrationRequest request) {
         // check if email is already in use, if not: add user to database and send verification email
-        if(userService.getUser(request.getEmail()).isEmpty()) {
-            User newUser = new User(request.getEmail(), request.getStudentId(), request.getFirstName(),
-                    request.getLastName(), request.getPassword(), "student", "");
-            userService.addUser(newUser);
+        if(unverifiedUserService.getUnverifiedUser(request.getEmail()).isEmpty()
```

Comment:
the service name **unverifiedUserService** is misleading. 
why it's prefixed with ``un``

Reply:
To keep naming convention as this class work with the table of unverified users in database

![Screenshot from 2022-12-16 22-51-19](https://user-images.githubusercontent.com/80472863/208186960-09503809-e682-4d84-9886-7b3b436f3c13.png)

Reply:
ok good job